src/pykeen/models/nbase.py

```diff
@@ -98,20 +98,40 @@ def _reset_parameters_(self):  # noqa: D401
             for i, p_id in enumerate(uninitialized_parameters, start=1):
                 logger.debug("[%3d] Parents to blame: %s", i, parents.get(p_id))
 
-    def _instantiate_default_regularizer(self, **kwargs) -> Optional[Regularizer]:
-        """Instantiate the regularizer from this class's default settings.
+    def _instantiate_regularizer(
+        self,
+        regularizer: HintOrType[Regularizer],
+        regularizer_kwargs: OptionalKwargs = None,
+    ) -> Optional[Regularizer]:
+        """
+        Instantiate a regularizer using the default if None is provided.
 
-        :param kwargs: Additional keyword arguments to be passed through to the ``__init__()`` function of the
-            default regularizer, if one is set.
+        The following precendence order is used:
 
-        :returns: If the default regularizer is None, None is returned.
-        """
-        if self.regularizer_default is None:
-            return None
+        1. If the passed regularizer is not None, use it
+        2. If the regularizer is None, use the default regularizer. In this case, the default kwargs will be used in
+           favour of provided ones.
+        3. If both, the regularizer and the default regularizer are None, return None.
 
-        _kwargs = dict(self.regularizer_default_kwargs or {})
-        _kwargs.update(kwargs)
-        return self.regularizer_default(**_kwargs)
+        :param regularizer:
+            the regularizer, or a hint thereof
+        :param regularizer_kwargs:
+            additional keyword-based parameters passed to the regularizer upon instantiation
+
+        :return:
+            the regularizer instance.
+        """
+        if regularizer is None:
+            regularizer = self.regularizer_default
+            if regularizer_kwargs is not None:
+                logger.warning(
+                    f"No regularizer was provided, but regularizer_kwargs={regularizer_kwargs} is not None. "
+                    f"Will use the default regularizer={self.regularizer_default} with "
+                    f"regularizer_kwargs={self.regularizer_default_kwargs}. If you want the explicitly provided "
+                    f"kwargs to be used, explicitly provide regularizer={self.regularizer_default} instead of None."
+                )
+            regularizer_kwargs = self.regularizer_default_kwargs
```

Comment:
shouldn't this back-fill any missing arguments with the defaults?

Reply:
I thought this would be the safer choice: If you *implicitly* use the default regularizer (by `regularizer=None`), you get the default kwargs. If you want something different, you can still *explicitly* pass the default and some kwargs

Reply:
Looking at https://github.com/pykeen/pykeen/pull/954, we might need to have more than one default regularizer, so maybe this methd might be converted to a function soon. I also added some comments about better supporting regularizer inference at `ERModel` level in some of the update to ER-Model PRs. 😅